code/controllers/subsystem/gamemaster.dm

```diff
@@ -0,0 +1,343 @@
+var/datum/subsystem/gamemaster/SSgamemaster
+
+/datum/subsystem/gamemaster
+	name = "Game Master"
+	wait = 100
+	var/probability = 0
+	var/first_announcement = FALSE
+	var/list/z1_areas = list()
+	var/list/event_list = list("bwoink", "singuloth", "el_dorado", "el_honkado", "privacy_windows", "give_guns", "ai_cameras", "ai_cameras",\
+							"communism", "knockdown", "change_dir", "fakespace", "make_antag", "mecha_warfare", "clumsy", "dwarves_n_giants",\
+							"investigate", "shuttle")
+
+/datum/subsystem/gamemaster/New()
+	NEW_SS_GLOBAL(SSgamemaster)
+
+/datum/subsystem/gamemaster/Initialize()
+	z1_areas = get_areas_in_z(1)
+	z1_areas |= get_areas(/area/hallway/primary/port)
+	..()
+
+/datum/subsystem/gamemaster/fire()
+	if(!first_announcement)
+		first_announcement = TRUE
+		gamemaster_announce("Let's all have fun together.")
+
+	if(prob(probability))
+		switch(pick_n_take(event_list))
+			if("bwoink")
+				bwoink()
+			if("singuloth")
+				singuloth()
+			if("el_dorado")
+				el_dorado()
+			if("el_honkado")
+				el_honkado()
+			if("privacy_windows")
+				privacy_windows()
+			if("give_guns")
+				give_guns()
+			if("ai_cameras")
+				ai_cameras()
+			if("communism")
+				communism()
+			if("knockdown")
+				knockdown()
+			if("change_dir")
+				change_dir()
+			if("fakespace")
+				fakespace()
+			if("make_antag")
+				make_antag()
+			if("mecha_warfare")
+				mecha_warfare()
+			if("clumsy")
+				clumsy()
+			if("dwarves_n_giants")
+				dwarves_n_giants()
+			if("investigate")
+				investigate()
+			if("shuttle")
+				shuttle()
```

Comment:
you could replace this switch case with call()() if you made the list have the actual procs